app/ui/shared/widgets/search.jsx

```diff
@@ -43,6 +43,8 @@ class Search extends Component {
 
     this.state = {
       interaction: INTERACTION_TYPES.IDLE,
+      focused: false,
+      focusByMouse: false,
```

Comment:
Instead of `focusByMouse`, it'd probably be better to use an interaction type (see INTERACTION_TYPES) so that the input's value doesn't change when the user interacts with it and the redux store decides it's a good idea to update it (even with the same value).